### PR TITLE
feat: Enable links in bio

### LINKED ISSUE #10237: Enable links in bio
It is a small detail, but a nice feature if we plan to push farcaster 

it would be nice if the links in the bio are also links on koda 
image

image

same with probably twitter links etc. 
image

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #10237

## Needs QA check

- @kodadot/qa-guild please review
- 
#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2024-05-29 at 09 59 06@2x

### CODE REVIEW

**Comment by hassnian** [comment]
> can we make it not affect markdown tags?
> 
> image

pls try again

**Review by Jarsen136 — APPROVED** [review]

**Review by vikiival — APPROVED** [review]
TL;DR 

parsing desc and checking for `/channle` (eg. `/koda`) or `@user` (e.g @vikiival)

cc @exezbcz I think it should be done on the create profile level, it would break markdown links imo

**Review by vikiival — CHANGES_REQUESTED** [review]
as pointed in comment above 

cc @daiagi

**Comment by vikiival** [comment]
> @vikiival how?

This PR

https://deploy-preview-10337--koda-canary.netlify.app/ahp/u/14BZFbYEGoWWPjbbZUdLZ1TqDtdXvdeCy3R3t4QXqJmS91Dx

Screenshot 2024-06-05 at 12 51 38


Prod:

Screenshot 2024-06-05 at 12 52 21

**Comment by vikiival** [comment]
TL;DR of your changes?

**Review by Jarsen136 — APPROVED** [review]

**Comment by hassnian** [comment]
> TL;DR of your changes?

Links are now added on profile save

**Comment by hassnian** [comment]
> > TL;DR of your changes?
> 
> Links are now added on profile save

@prury can you test again, [this](https://github.com/kodadot/nft-gallery/pull/10337#issuecomment-2149495373) should be fixed  , thanks

**Comment by prury** [comment]
> > > TL;DR of your changes?
> > 
> > 
> > Links are now added on profile save
> 
> @prury can you test again, [this](https://github.com/kodadot/nft-gallery/pull/10337#issuecomment-2149495373) should be fixed , thanks

seems good now

**Comment by vikiival** [comment]
Screenshot 2024-06-11 at 11 33 17

**Comment by vikiival** [comment]
Works as intented.
But I have a problem:

I use create profile (manual), inserted fan of `/koda` it was automatically translated to warpcast link. 

My question is: Should it be enabled only in case when you use import from farcaster ? 

cc @exezbcz

**Comment by vikiival** [comment]
Screenshot 2024-06-11 at 11 41 21


also thinking why you do not translate it to markdown links? rather than html? 

```
<a href="https://warpcast.com/~/channel/koda" target="_blank" rel="nofollow noopener noreferrer">/koda</a>
```

vs 

```
[/koda](https://warpcast.com/~/channel/koda)
```

**Review by vikiival — CHANGES_REQUESTED** [review]
as per exez's comment above

**Comment by hassnian** [comment]
> also thinking why you do not translate it to markdown links? rather than html?

@vikiival so they open in a new tab , can change it if it's not important.

**Comment by hassnian** [comment]
> > My question is: Should it be enabled only in case when you use import from farcaster ?
> 
> /koda should be translated only when importing from farcaster
> 
> as /koda can mean something else on koda itself in the future

fixed

**Comment by vikiival** [comment]
> > also thinking why you do not translate it to markdown links? rather than html?
> 
> 
> 
> @vikiival so they open in a new tab , can change it if it's not important.

Every link on profiles are markdown based. I prefer to have it unified

**Comment by hassnian** [comment]
> Every link on profiles are markdown based. I prefer to have it unified

done

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
> @hassnian can you check profile creation trough farcaster using this PR? i tried testing here and it won't work on canary it goes normally.

cc @hassnian

**Comment by hassnian** [comment]
> > @hassnian can you check profile creation trough farcaster using this PR? i tried testing here and it won't work on canary it goes normally.
> 
> cc @hassnian

checking
